static/js/ui.js

```diff
@@ -220,13 +218,10 @@ function scroll_finished() {
             pointer.suppress_scroll_pointer_update = false;
         }
         floating_recipient_bar.update();
-        if (message_viewport.scrollTop() === 0 &&
-            ui.have_scrolled_away_from_top) {
-            ui.have_scrolled_away_from_top = false;
+        if (message_viewport.scrollTop() === 0) {
             message_fetch.load_more_messages(current_msg_list);
-        } else if (!ui.have_scrolled_away_from_top) {
-            ui.have_scrolled_away_from_top = true;
         }
+
```

Comment:
So, reading the code we're removing code, I imagine its intent was to avoid having a "fetch more messages from the server" loop at the top of the message feed.  Do you understand what mechanism should be preventing that now?  

I imagine the mechanism would be that `load_more_enabled` gets set to `true` when you narrow, and set back to `false` when you fetch back a number of message above that's less than you asked for.  

Also, you missed a reference to this varaible in `message_fetch.js`.